### ISSUE TITLE
Pages resource suddenly causing 403 failures

### ISSUE THREAD

**Issue by dhess** [integrations/terraform-provider-github#1594]
Hi, we have a private repo with a public GitHub Pages. This setup has been working fine with this Terraform provider for months, until sometime yesterday, when it started failing (100% repeatably) with the following error during plans and applies:

```
Error: GET https://api.github.com/repos/hackworthltd/xxxx/pages: 403 Resource not accessible by integration []
```

We had been setting the `pages` attribute of this repo in our Terraform, so I removed the `pages` that configuration, manually removed the repository from the statefile, and re-imported it. The import failed with the same 403.

Next, I deleted the GitHub Pages (deleted the deployed site & set the GitHub Pages deployment branch to "None"), then re-imported the repo again into our Terraform state. This time it worked, and our Terraform applies started working again, as well.

I then re-published the repo's GitHub Pages, and now the 403's have returned.

Any ideas? The only thing I can think of that changed on our end is that we started publishing a package in this repo around the same time that the failures began, but I can't imagine why that would matter.

**Comment by mhodzic93** [comment]
I can confirm that I'm seeing the exact same behavior using the latest github terraform provider version, below is the debug output
`2023-03-16T16:36:51.316-0700 [DEBUG] provider.terraform-provider-github_v5.18.3:  "message": "Resource not accessible by integration",
2023-03-16T16:36:51.316-0700 [DEBUG] provider.terraform-provider-github_v5.18.3:  "documentation_url": "https://docs.github.com/rest/pages#get-a-github-pages-site"
2023-03-16T16:36:51.316-0700 [DEBUG] provider.terraform-provider-github_v5.18.3: }
`
Our setup has been working as well for more than a year now and just suddenly started failing. I've tried multiple previous versions and all of them are failing. I've looked through the change logs for github and terraform and haven't noticed anything so it's quite puzzling as to how this has become an issue out of nowhere. Hopefully someone can explain what the cause of this issue is.
